src/js/deepLinks.ts

```diff
@@ -29,9 +30,63 @@ for (let dr in deepRoutes) deepRouter.add(dr, deepRoutes[dr]);
 
 export default {
   init() {
-    // open games from external links with url scheme
-    window.handleOpenURL = function(url: string) {
+    // open links that follow the custom url scheme
+      window.handleOpenURL = function(url: string) {
       setTimeout(() => deepRouter.run(url.replace(/^lichess:\/\//, '')), 0);
     };
+
+    // open normal Lichess links
+    const universalLinks = window.universalLinks;
+    universalLinks.subscribe('analysis', () => router.set('/analyse'));
+    universalLinks.subscribe('challenge', (eventData: EventData) => router.set('/challenge/' + eventData.path.split('/').pop()));
+    universalLinks.subscribe('editor', () => router.set('/editor'));
+    universalLinks.subscribe('inbox', () => router.set('/inbox'));
+    universalLinks.subscribe('inboxNew', () => router.set('/inbox/new'));
+    universalLinks.subscribe('players', () => router.set('/players'));
+    universalLinks.subscribe('tournamentDetail', (eventData: EventData) => router.set('/tournament/' + eventData.path.split('/').pop()));
+    universalLinks.subscribe('tournamentList', () => router.set('/tournament'));
+    universalLinks.subscribe('training', () => router.set('/training'));
+    universalLinks.subscribe('trainingProblem', handleTrainingProblem);
+    universalLinks.subscribe('tv', () => router.set('/tv'));
+    universalLinks.subscribe('tvChannel', (eventData: EventData) => router.set('/tv/' + eventData.path.split('/').pop()));
+    universalLinks.subscribe('userVariantProfile', handleVariantProfile);
+    universalLinks.subscribe('userProfile', (eventData: EventData) => router.set('/@/' + eventData.path.split('/').pop()));
+    universalLinks.subscribe('other', handleOther);
   }
 };
+
+function handleVariantProfile (eventData: EventData) {
+  const pieces = eventData.path.split('/');
+  const uid = pieces[2];
+  const variant = pieces[4];
+  router.set('/@/' + uid + '/' + variant + '/perf');
+}
+
+function handleTrainingProblem (eventData: EventData) {
+  const pieces = eventData.path.split('/');
+  const problem = pieces[2];
+  if (problem === 'coordinate') {
```

Comment:
The issue is that the `universalLinks` cordova plugin does not have regex support. We can only match paths against a simple `*` wildcard. Because games, which we definitely want to give the user the ability to direct to the app, are of the form `https://en.lichess.org/[8 letters or numbers]`, we have to handle the case of `https://*.lichess.org/*`, test to see if it's a game with javascript regex matching code, and then direct back out to a browser if not.
The same applies the training. We have to match `https://*.lichess.org/training/*` and then direct the user back out to the web browser if `*` turns out to be `coordinate` rather than some problem number.